lib/TextArea.js

```diff
@@ -0,0 +1,53 @@
+CustomElement = require('./CustomElement')
+
+CustomElement.create({
+  tagName: 'nri-textarea-v3',
+
+  initialize: function() {
+    this._autoresize = false
+  },
+
+  onConnect: function() {
+    this._textarea = this.querySelector('textarea')
+    this._updateListener()
+  },
+
+  properties: {
+    autoresize: {
+      get: function() {
+        return this._autoresize
+      },
+      set: function(value) {
+        if (this._autoresize === value) return
+        this._autoresize = value
+        this._updateListener()
+      }
+    }
+  },
+
+  methods: {
+    _updateListener: function() {
+      if (this._autoresize) {
+        this._textarea.addEventListener('input', this._resize)
+        this._resize()
+      } else {
+        this._textarea.removeEventListener('input', this._resize)
+      }
+    },
+
+    _resize: function() {
```

Comment:
I made changes to the resize logic here: https://github.com/NoRedInk/noredink-ui/pull/63/files#diff-b310ebe2d8a713781bb80d92660737f1R51 (which are missing from this version)

- look at the computed min-height in addition, as it's possible min-height is specified through a class
- reset to minHeight instead of `auto`, as `auto` insisted on adding some padding to `scrollHeight` when minHeight is only a single line's worth
  - set transition to none as I couldn't figure out why the textarea started doing animation after changing the above when it didn't before

Reply:
included, thank you!